### DISCUSSION TITLE
BLE Hub to PC 51515

### DISCUSSION THREAD

**Discussion by AGarves** [pybricks/support#518] [Q&A]
Thanks for the help in advance! Been scratching my head for a good amount of time

I've seen several successfull attempts on connecting with other hubs via pc #181 and #470. The last one in particular seems to be the library i installed through pip on my envirement. Assuming that its not nessecary to have pybricks installed on the hub to connect with the pybricks library on my vsc, i would simply need to use something like:

```
from pybricksdev.connections import PybricksHub
from pybricksdev.ble import find_device

async def main():
    hub = PybricksHub()
    device = await find_device()
    await hub.connect(device)
```

But i don't get any successful connection even if i run with:
```device = await find_device("LEGO Hub") ```
or when running Pybricks on the Hub:
``` device = await find_device("Pybricks Hub") ```

 Am i missing something here? My PC runs BLE. Also if i should add the UUID, where can i find mine. What about mac adress and the timeout in the docs?

Furthermore when using run from asyncio, do i need to specify the programm on the hub as what print(__ file __) returns or should i go with the name (in case of pybricks on the hub: main.py)

-Garves

**Comment by laurensvalk** [comment]
>  Assuming that its not nessecary to have pybricks installed on the hub to connect with the pybricks library on my vsc

Pybricksdev can only be used to run programs via BLE if the hub runs Pybricks firmware.

What would you like to achieve exactly? Depending on what you would like to do, there may be alternate solutions.

**Comment by AGarves** [comment]
Very fast reply, thank you very much. 

I would like to use a PS4Controller and drive my mecanum-wheel vehicle with its sticks. My plan was to track the sticks' x and y coordinates with pygame on my pc and send the results of their associated angle to my hub. The hub would process that angle into motor speed. The whole idea would be to not hardcode with with the blocks and mindstorms controller streaming since i want to use a method which can calculate every motorspeed needed for any given angle. Tracking PS4Controller (no BLE) would use alot of memory. Pygame is obviously no default .mpy library.

Thank you again for the help
-Garves
